### PR TITLE
Feature/user database ( User Setup)

### PR DESCRIPTION
**Pull request by Omarjabari007**
Hey, I created the database using docker and used Dbeaver app to show the results of the database ..., I only manage the user entity for now ,, If any method realted to the user entity Is not included in the code that because it's depended on other teammate member work, So I will wait my teammate to finish his job to complete .., I already did the product part also but I need to move it into the database .. and that will be the next pr hopefully .

### CODE REVIEW

**Comment by Hatem29** [comment]
approved in my end

**Comment by Mohammed-samaheen** [comment]
> approved in my end

That's not how we approve.

**Comment by Hatem29** [comment]
I do not have the option to approve and Omar needed my feedback on it so I had to do it this way.

File: .gitignore
Comment (Mohammed-samaheen) [inline .gitignore]: docker compose file in gitignore!!!

File: app/api/exceptions/GlobalException.py
Comment (Mohammed-samaheen) [inline app/api/exceptions/GlobalException.py:1]: use the same file name convention.

File: app/api/routes/user.py
Comment (Mohammed-samaheen) [inline app/api/routes/user.py]: we should never return error details to the user. This includes the Exception content, in this case, it's the `e`

File: app/api/routes/user.py
Comment (Mohammed-samaheen) [inline app/api/routes/user.py]: response?

File: app/api/services/user_service.py
Comment (Mohammed-samaheen) [inline app/api/services/user_service.py:59]: user can update their email

File: app/db/database.py
Comment (Mohammed-samaheen) [inline app/db/database.py]: This should be an environment valuable.

File: app/main.py
Comment (Mohammed-samaheen) [inline app/main.py]: Why did you move the routers to app/main.py?

File: app/models.py
Comment (Mohammed-samaheen) [inline app/models.py]: The default value for the update date should be null.

File: app/models.py
Comment (Mohammed-samaheen) [inline app/models.py]: What does DTO mean? Please make the name clearer.

File: app/models.py
Comment (Mohammed-samaheen) [inline app/models.py]: UserUpdateRequest

File: app/schemas/user.py
Comment (Mohammed-samaheen) [inline app/schemas/user.py]: why do we have user model in app/models.py and app/schemas/user.py?

**Review by Mohammed-samaheen — CHANGES_REQUESTED** [review]
- Next time, try to make the PR task specific and avoid including multiple tasks in the same PR.
- All user endpoints should be updated when you add the security part.

- I didn't find any comments from the other team members in this PR, so I will assume that Omar did all the work. (even though we shouldn't be sharing the same branch).

- Anyone can add their approval; you don't need permission to do that. So next time, I won't review the PR until it's approved by at least one member of the team.

File: .gitignore
Comment (Omarjabari007) [reply .gitignore]: I wanted to post it as a standalone  PR thing , dw I fixed it

File: app/api/exceptions/GlobalException.py
Comment (Omarjabari007) [reply app/api/exceptions/GlobalException.py:1]: Done

File: app/api/routes/user.py
Comment (Omarjabari007) [reply app/api/routes/user.py]: Done .

File: app/api/routes/user.py
Comment (Omarjabari007) [reply app/api/routes/user.py]: Sry for that I know we must not return any response on delete operation , I think i did it twice to return sucessfully deleted , I will fix it

File: app/api/services/user_service.py
Comment (Omarjabari007) [reply app/api/services/user_service.py:59]: Done .

File: app/db/database.py
Comment (Omarjabari007) [reply app/db/database.py]: Ok

**Review by Hatem29 — APPROVED** [review]

File: app/models.py
Comment (Omarjabari007) [reply app/models.py]: Sry, It mean Data transfer model ( same mean for DataResponse or DataCreate or DataUpdate ) but it's the global meaning .

File: app/models.py
Comment (Omarjabari007) [reply app/models.py]: I did all of the things , but not this  , can i get a hint for it because make the value null will lead to a  lot of bugs .

Comment (Mohammed-samaheen) [reply app/models.py]: What are the bugs?

File: app/db/database.py
Comment (Mohammed-samaheen) [inline app/db/database.py]: Please use pydintic  setting to load the environment variables.

File: app/models.py
Comment (Mohammed-samaheen) [reply app/models.py]: if there is any bug regarding having the update date as null, you should fix it

File: docker-compose.yml
Comment (Mohammed-samaheen) [inline docker-compose.yml]: please add the required environment variables to the readme file or use a default value in the compose file

File: docker-compose.yml
Comment (Mohammed-samaheen) [inline docker-compose.yml]: Deprecated: no need for this line.

File: .gitignore
Comment (Mohammed-samaheen) [inline .gitignore]: we use this folder for code owners  and implementation of workflows

**Review by Mohammed-samaheen — CHANGES_REQUESTED** [review]

File: docker-compose.yml
Comment (Omarjabari007) [reply docker-compose.yml]: Done

File: app/models.py
Comment (Omarjabari007) [reply app/models.py]: Done.
the problem is the database can't accept a null value for update, I tried everything, and the solution was to alter the table manually, I did from Dbeaver SQL commands.

File: app/db/database.py
Comment (Omarjabari007) [reply app/db/database.py]: Done

File: app/models.py
Comment (Mohammed-samaheen) [reply app/models.py]: This is most likely because you're using a Docker volume. Whenever you start the service, the table will already be present, and the CREATE TABLE IF NOT EXISTS will only create the table if it doesn't exist. So, next time, you can delete the volume folder or not use volume for now.

File: .gitignore
Comment (Mohammed-samaheen) [reply .gitignore]: please fix this

File: docker-compose.yml
Comment (Mohammed-samaheen) [inline docker-compose.yml:12]: When you run the service, the tables and all the data will be stored in the new_postgres_data folder. If you want to recreate everything from scratch, you can delete that folder

File: docker-compose.yml
Comment (Mohammed-samaheen) [inline docker-compose.yml:6]: There won't be an issue explicitly setting the value in the Docker Compose file, because we only use it locally. In production, we pass different values.

File: app/main.py
Comment (Mohammed-samaheen) [inline app/main.py]: why did you remove the prefix?

**Review by Mohammed-samaheen — CHANGES_REQUESTED** [review]

File: docker-compose.yml
Comment (Omarjabari007) [reply docker-compose.yml:12]: Thanks for this, I was searching about it

File: .gitignore
Comment (Omarjabari007) [reply .gitignore]: sorry I fixed it but I forget to commit it .

File: docker-compose.yml
Comment (Omarjabari007) [reply docker-compose.yml:6]: Got it , but it's ok to leave it like this ?

File: app/main.py
Comment (Omarjabari007) [reply app/main.py]: Done

**Review by Mohammed-samaheen — APPROVED** [review]

**Pull request merged**
